testnet/Dockerfile.debug

```diff
@@ -0,0 +1,46 @@
+# BUILD IMAGE ---------------------------------------------------------
+
+FROM rust:1.85.1-slim-bookworm AS builder
+
+# Install dependencies needed for building RocksDB and etcd.
+RUN apt-get update && apt-get install -yq \
+    git gcc g++ clang libssl-dev pkg-config ca-certificates curl cmake ninja-build
+
+COPY . /nomos
+
+WORKDIR /opt
+
+RUN git clone https://github.com/risc0/risc0.git && \
+    cd risc0 && \
+    cargo install --path rzup && \
+    rzup build rust --tag-or-commit 4d91de4e48198da2e33413efdcd9cd2cc0c46688 && \
+    git checkout origin/release-2.0 && \
+    cargo install --path risc0/cargo-risczero && \
+    rm -rf /usr/local/cargo/registry /usr/local/cargo/git && \
+    rm -rf /root/.risc0/tmp && \
+    rm -rf * && \
+    cd /nomos && \
+    cargo build && \
```

Comment:
I think it would also speed up compilation if we compile only what we need. So I think it's fine to build all and only the necessary binaries here. We could then use this base image in the future to build one Docker image for each binary, if needed, re-using the output of this single image.

Reply:
Great idea. I'll look at this in coming weeks.

Reply:
So this will be tackled in a separate PR?

Reply:
> So this will be tackled in a separate PR?

Yes, I suggest to have this optimized in the next evolution. And perhaps, (not guaranteed) Risc0 team will make it happen, as I proposed here https://discord.com/channels/953703904086994974/1331206636139970621